Javassist does not handle interface duplication method correctly

### Describe the bug
Javassist reports an error when encountering duplicate method: "duplicate method".
When I change default implementation of the proxy  to Bytebuddy, the service starts successfully.

### Expected behavior
Duplicate methods should only generate one method.

### Actual behavior
```log
Exception in thread "main" com.alipay.sofa.rpc.core.exception.SofaRpcRuntimeException: RPC-010080001: 使用 [javassist] 构造代理出现错误. 
	at com.alipay.sofa.rpc.proxy.javassist.JavassistProxy.getProxy(JavassistProxy.java:149)
	at com.alipay.sofa.rpc.proxy.ProxyFactory.buildProxy(ProxyFactory.java:50)
	at com.alipay.sofa.rpc.bootstrap.DefaultConsumerBootstrap.refer(DefaultConsumerBootstrap.java:156)
	at com.alipay.sofa.rpc.config.ConsumerConfig.refer(ConsumerConfig.java:955)
	at me.s2marine.QuickStartClient.main(QuickStartClient.java:14)
Caused by: javassist.bytecode.DuplicateMemberException: duplicate method: getType in me.s2marine.QuickStartClient$HelloService_proxy_0
	at javassist.bytecode.ClassFile.testExistingMethod(ClassFile.java:685)
	at javassist.bytecode.ClassFile.addMethod(ClassFile.java:660)
	at javassist.CtClassType.addMethod(CtClassType.java:1498)
	at com.alipay.sofa.rpc.proxy.javassist.JavassistProxy.getProxy(JavassistProxy.java:125)
	... 4 more
```

### Steps to reproduce

### Minimal yet complete reproducer code (or GitHub URL to code)

```java
public class QuickStartClient {

    public static void main(String[] args) {
        ConsumerConfig<HelloService> consumerConfig = new ConsumerConfig<HelloService>()
                .setInterfaceId(HelloService.class.getName()) // Specify the interface
                .setProtocol("bolt") // Specify the protocol.setDirectUrl
                .setDirectUrl("bolt://127.0.0.1:12200"); // Specify the direct connection address

        // Generate the proxy class
        HelloService helloService = consumerConfig.refer();
    }

    public interface HelloService extends SubInterface1, SubInterface2 {
        String sayHello(String string);
    }

    interface SubInterface1 {
        String getType();

        String someMethod1();
    }

    interface SubInterface2 {
        String getType();

        String someMethod2();
    }
}
```

### Environment

- SOFARPC version: 5.11.1
- JVM version (e.g. `java -version`): 1.8.0_392
- OS version (e.g. `uname -a`): Linux archlinux-r9000p-s2marine 6.6.8-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 21 Dec 2023 19:01:01 +0000 x86_64 GNU/Linux
- Maven version: Apache Maven 3.8.7
- IDE version: Intellij IDEA 2023.3.2

--- a/core-impl/proxy/src/main/java/com/alipay/sofa/rpc/proxy/javassist/JavassistProxy.java
+++ b/core-impl/proxy/src/main/java/com/alipay/sofa/rpc/proxy/javassist/JavassistProxy.java
@@ -42,8 +42,10 @@ import java.lang.reflect.Field;
 import java.lang.reflect.Method;
 import java.lang.reflect.Modifier;
 import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.atomic.AtomicInteger;
 
@@ -154,12 +156,22 @@ public class JavassistProxy implements Proxy {
         Method[] methodAry = interfaceClass.getMethods();
         StringBuilder sb = new StringBuilder(512);
         int mi = 0;
+        // Track method signatures to skip duplicate methods from multiple parent interfaces.
+        // When an interface extends multiple parent interfaces that declare the same method,
+        // interfaceClass.getMethods() returns duplicates, causing Javassist to throw
+        // DuplicateMemberException when adding the same method twice.
+        // See: https://github.com/sofastack/sofa-rpc/issues/1384
+        Set<String> addedMethodSignatures = new HashSet<String>();
         for (Method m : methodAry) {
-            mi++;
             if (Modifier.isNative(m.getModifiers()) || Modifier.isFinal(m.getModifiers()) ||
                 Modifier.isStatic(m.getModifiers())) {
                 continue;
             }
+            // Skip duplicate methods inherited from multiple parent interfaces
+            if (!addedMethodSignatures.add(buildMethodSignature(m))) {
+                continue;
+            }
+            mi++;
             Class<?>[] mType = m.getParameterTypes();
             Class<?> returnType = m.getReturnType();
 
@@ -283,6 +295,28 @@ public class JavassistProxy implements Proxy {
     @Override
     public Invoker getInvoker(Object proxyObject) {
         return parseInvoker(proxyObject);
+    }
+
+    /**
+     * Builds a unique method signature string in the format: methodName(paramType1,paramType2,...)
+     * Used to deduplicate methods inherited from multiple parent interfaces that declare the same method.
+     * Without deduplication, Javassist throws DuplicateMemberException when the same method is added twice.
+     *
+     * @param method the method to build a signature for
+     * @return the method signature string
+     */
+    private String buildMethodSignature(Method method) {
+        StringBuilder signature = new StringBuilder(method.getName());
+        signature.append('(');
+        Class<?>[] parameterTypes = method.getParameterTypes();
+        for (int i = 0; i < parameterTypes.length; i++) {
+            if (i > 0) {
+                signature.append(',');
+            }
+            signature.append(parameterTypes[i].getName());
+        }
+        signature.append(')');
+        return signature.toString();
     }
 
     /**

--- a/core-impl/proxy/src/main/java/com/alipay/sofa/rpc/proxy/javassist/JavassistProxy.java
+++ b/core-impl/proxy/src/main/java/com/alipay/sofa/rpc/proxy/javassist/JavassistProxy.java
@@ -161,14 +161,14 @@ public class JavassistProxy implements Proxy {
         // interfaceClass.getMethods() returns duplicates, causing Javassist to throw
         // DuplicateMemberException when adding the same method twice.
         // See: https://github.com/sofastack/sofa-rpc/issues/1384
-        Set<String> addedMethodSignatures = new HashSet<String>();
+        Set<String> seenSignatures = new HashSet<>();
         for (Method m : methodAry) {
-            if (Modifier.isNative(m.getModifiers()) || Modifier.isFinal(m.getModifiers()) ||
+            if (m.isBridge() || Modifier.isNative(m.getModifiers()) || Modifier.isFinal(m.getModifiers()) ||
                 Modifier.isStatic(m.getModifiers())) {
                 continue;
             }
             // Skip duplicate methods inherited from multiple parent interfaces
-            if (!addedMethodSignatures.add(buildMethodSignature(m))) {
+            if (!seenSignatures.add(buildMethodSignature(m))) {
                 continue;
             }
             mi++;
@@ -298,16 +298,20 @@ public class JavassistProxy implements Proxy {
     }
 
     /**
-     * Builds a unique method signature string in the format: methodName(paramType1,paramType2,...)
+     * Builds a unique method signature string in the format: returnType methodName(paramType1,paramType2,...)
      * Used to deduplicate methods inherited from multiple parent interfaces that declare the same method.
      * Without deduplication, Javassist throws DuplicateMemberException when the same method is added twice.
+     *
+     * <p>The return type is included to correctly handle covariant return types, where two methods
+     * may share the same name and parameter types but differ in return type (e.g. bridge methods).
      *
      * @param method the method to build a signature for
      * @return the method signature string
      */
     private String buildMethodSignature(Method method) {
-        StringBuilder signature = new StringBuilder(method.getName());
-        signature.append('(');
+        StringBuilder signature = new StringBuilder();
+        signature.append(method.getReturnType().getName()).append(' ');
+        signature.append(method.getName()).append('(');
         Class<?>[] parameterTypes = method.getParameterTypes();
         for (int i = 0; i < parameterTypes.length; i++) {
             if (i > 0) {

--- a/core-impl/proxy/src/test/java/com/alipay/sofa/rpc/proxy/javassist/DiamondHelloService.java
+++ b/core-impl/proxy/src/test/java/com/alipay/sofa/rpc/proxy/javassist/DiamondHelloService.java
@@ -1,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.rpc.proxy.javassist;
+
+/**
+ * Service interface that extends two parent interfaces declaring the same method (diamond inheritance).
+ * Used to verify that JavassistProxy correctly deduplicates inherited methods.
+ *
+ * @see <a href="https://github.com/sofastack/sofa-rpc/issues/1384">#1384</a>
+ */
+public interface DiamondHelloService extends DiamondSubInterface1, DiamondSubInterface2 {
+
+    String sayHello(String message);
+}

--- a/core-impl/proxy/src/test/java/com/alipay/sofa/rpc/proxy/javassist/DiamondSubInterface1.java
+++ b/core-impl/proxy/src/test/java/com/alipay/sofa/rpc/proxy/javassist/DiamondSubInterface1.java
@@ -1,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.rpc.proxy.javassist;
+
+/**
+ * Sub interface for diamond inheritance test.
+ * Declares getType() which is also declared in DiamondSubInterface2.
+ *
+ * @see <a href="https://github.com/sofastack/sofa-rpc/issues/1384">#1384</a>
+ */
+public interface DiamondSubInterface1 {
+
+    String getType();
+
+    String someMethod1();
+}

--- a/core-impl/proxy/src/test/java/com/alipay/sofa/rpc/proxy/javassist/DiamondSubInterface2.java
+++ b/core-impl/proxy/src/test/java/com/alipay/sofa/rpc/proxy/javassist/DiamondSubInterface2.java
@@ -1,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.rpc.proxy.javassist;
+
+/**
+ * Sub interface for diamond inheritance test.
+ * Declares getType() which is also declared in DiamondSubInterface1.
+ *
+ * @see <a href="https://github.com/sofastack/sofa-rpc/issues/1384">#1384</a>
+ */
+public interface DiamondSubInterface2 {
+
+    String getType();
+
+    String someMethod2();
+}

--- a/core-impl/proxy/src/test/java/com/alipay/sofa/rpc/proxy/javassist/JavassistProxyTest.java
+++ b/core-impl/proxy/src/test/java/com/alipay/sofa/rpc/proxy/javassist/JavassistProxyTest.java
@@ -17,6 +17,8 @@
 package com.alipay.sofa.rpc.proxy.javassist;
 
 import com.alipay.sofa.rpc.core.request.SofaRequest;
+import com.alipay.sofa.rpc.core.response.SofaResponse;
+import com.alipay.sofa.rpc.invoke.Invoker;
 import com.alipay.sofa.rpc.log.Logger;
 import com.alipay.sofa.rpc.log.LoggerFactory;
 import com.alipay.sofa.rpc.proxy.AbstractTestClass;
@@ -105,4 +107,30 @@ public class JavassistProxyTest {
         Assert.assertTrue(error);
     }
 
+    /**
+     * Tests that JavassistProxy correctly handles diamond interface inheritance
+     * where multiple parent interfaces declare the same method signature.
+     * Before the fix, this would throw DuplicateMemberException.
+     *
+     * @see <a href="https://github.com/sofastack/sofa-rpc/issues/1384">#1384</a>
+     */
+    @Test
+    public void testDiamondInheritanceProxy() {
+        JavassistProxy proxy = new JavassistProxy();
+
+        Invoker mockInvoker = request -> {
+            SofaResponse response = new SofaResponse();
+            response.setAppResponse("mock-" + request.getMethodName());
+            return response;
+        };
+
+        DiamondHelloService service = proxy.getProxy(DiamondHelloService.class, mockInvoker);
+        Assert.assertNotNull(service);
+
+        // Verify all methods from both sub-interfaces and the service itself are callable
+        Assert.assertEquals("mock-sayHello", service.sayHello("test"));
+        Assert.assertEquals("mock-getType", service.getType());
+        Assert.assertEquals("mock-someMethod1", service.someMethod1());
+        Assert.assertEquals("mock-someMethod2", service.someMethod2());
+    }
 }

